packages/manager/modules/pci/src/projects/project/kubernetes/details/service/edit-name/edit-name.controller.js

```diff
@@ -9,8 +10,16 @@ export default class {
 
   $onInit() {
     this.isEditing = false;
+    this.name = this.cluster && this.cluster.name;
 
-    this.name = this.cluster.name;
+    this.constants = {
+      nameMaxLength: NAME_INPUT_CONSTRAINTS.MAX_LENGTH,
+      pattern: NAME_INPUT_CONSTRAINTS.PATTERN,
+    };
+  }
+
+  isEditDisabled() {
+    return !(this.name && this.name.match(this.constants.pattern));
```

Comment:
return !(this.name && this.name.match(NAME_INPUT_CONSTRAINTS.PATTERN));

Reply:
Done